routers/api/packages/generic/generic.go

```diff
@@ -18,8 +18,8 @@ import (
 )
 
 var (
-	packageNameRegex = regexp.MustCompile(`\A[A-Za-z0-9\.\_\-\+]+\z`)
-	filenameRegex    = packageNameRegex
+	packageNameRegex = regexp.MustCompile(`\A[-_+.A-Za-z0-9]+\z`)
+	filenameRegex    = regexp.MustCompile(`\A[-_+=:;.()\[\]{}~!@#$%^& A-Za-z0-9]+\z`)
```

Comment:
I would like not to have a file that starts or ends with whitespaces.
I have no problem with it in the middle, but it will cause problems at the beginning or end.
```suggestion
filenameRegex = regexp.MustCompile(`\A[-_+=:;.()\[\]{}~!@#$%^&A-Za-z0-9][-_+=:;.()\[\]{}~!@#$%^&A-Za-z0-9]*[-_+=:;.()\[\]{}~!@#$%^&A-Za-z0-9]+\z`)
```

Reply:
Additionally, should we perhaps disallow `$` to make life easier for linux users?

Reply:
`:` already makes windows users harder.

Reply:
> I would like not to have a file that starts or ends with whitespaces.

I guess it doesn't need to be a hard requirement. Just let users do what they want. At least, it doesn't harm Gitea.

Reply:
Yes, not Gitea, but end users.

Reply:
> Yes, not Gitea, but end users.

I couldn't agree. My philosophy is just letting end users do what they want, if it doesn't really cause problem.

If you think `$` causes problem, then there are more cases like `touch a #b`, then `#b` is comment.

I do not think it is worth to make things more complex than it should be.

Reply:
And one more case: `..`